### PR TITLE
Dutch accent

### PR DESCRIPTION
**Pull request by Bierkraan**
## About the PR
This RP adds a Dutch accent and you shoes will be replaced by a pair of wooden clogs if you sellect the trait. The clogs are not findable in any other way.

## Why's this needed?
There currently are only 7 accents in the game (more if you count the OwO's and some others) and having some more would be nice for a refreshment and RP.

Me and Yellow are making a few more languange that excludes stereotype words and focuses more on the phonemes, if you would like to help send one of us a message in discord.

### CODE REVIEW

**Comment by flrsh** [comment]
Can you give us some example sentences and what they would read like with the Dutch accent?

File: code/modules/medical/genetics/bioEffects/speech.dm [hunk 1]
```diff
@@ -106,6 +106,27 @@
...
 		return message
 
+/datum/bioEffect/speech/dutch
+	name = "Frontal Gyrus Alteration Type-NL"
+	desc = "Forces the language center of the subject's brain to construct sentences in the manner of a space dike builder."
+	id = "accent_dutch"
+	effectType = EFFECT_TYPE_DISABILITY
+	isBad = 1
+	msgGain = "You feel like putting on some wooden clogs!"
+	msgLose = "You suddenly lose intrest in wooden clogs."
```
Comment (BenLubar) [inline code/modules/medical/genetics/bioEffects/speech.dm:116]:
```suggestion
	msgLose = "You suddenly lose interest in wooden clogs."
```

File: code/procs/accents.dm [hunk 2]
```diff
@@ -1835,7 +1835,68 @@ var/list/zalgo_mid = list(
...
+	var/used = 0
+
+	switch(R.curr_char)
+
+		if("t")//dutch have problems with th
+			if(lowertext(R.next_char) == "h")
+				new_string = "t"
+				used = 2
+		if("T")
+			if(lowertext(R.next_char) == "H")
```
Comment (BenLubar) [inline code/procs/accents.dm:1853]: Might want to also have a case for `Th` as that occurs quite often in English.

File: code/obj/item/clothing/shoes.dm [hunk 3]
```diff
@@ -382,6 +382,25 @@
...
+	step_priority = STEP_PRIORITY_LOW
+	setupProperties()
+		..()
+		setProperty("meleeprot", 1)
+
+	get_desc(var/dist, var/mob/user)
+		if (user.traitHolder.hasTrait("dutch"))
+			. = "Remembers you of the homeland."
+		else
+			. = "A pair of shoes from someone far under sea level."
```
[the remark this answers is not in the document]
Comment (Bierkraan) [reply code/obj/item/clothing/shoes.dm:401]:
```suggestion
			+= "Remembers you of the homeland."
		else
			+= "A pair of shoes from someone far under sea level."
```
like this?

**Comment by Bierkraan** [comment]
> Can you give us some example sentences and what they would read like with the Dutch accent?

people = phieple, the = te, worked = workud

**Comment by pali6** [comment]
Example of the accent in action:

> Micheal Van Dyk says, "Heisenbee had done a very bad crime in te animal courtroom. He stung Albert te Monkey witout warning and tat made Albert very sad."
> 
> Micheal Van Dyk exclaims, "Albert te Monkey had asked te owl pholice, which was headed by Hooty McJudgementowl, to arrest Heisenbee, which is a very mean ting to do to a bee!"
> 
> Micheal Van Dyk says, "The owl jury stood silent as Heisenbee's lawyer, a fresh-faced monkey by te name of Tanhony and a voice te sound of horses running wild entered te courtroom."
> 
> Micheal Van Dyk says, "Heisenbee buzzed questioningly at te judge! The judge, his Honor Muggles screeched back Heisenbee's terrible and awful crimes! Tanhony asked Heisenbee to remain quiet."
> 
> Micheal Van Dyk says, "The mammalian Jury grew restless. Jones te cat hissed at te mouse, who was taking notes of te phroceedings. Tanhony gave his aphpheal to te animal Jury."
> 
> Micheal Van Dyk exclaims, "The broters gnome, Chomphski and Chomphski, were not convinced and voted Heisenbee guilty of attemphted murder! Which was very bad for Heisenbee!"
> 
> Micheal Van Dyk says, "The owls stood silent, which as we all know means tat tey knew in teir hearts of hearts tat Heisenbee was guilty."
> 
> Micheal Van Dyk says, "The mammalian jury was silent, having just convinced te cockroach tat Heisenbee was not like oter bugs and was in fact guilty."
> 
> Micheal Van Dyk says, "Heisenbee buzzed in phanic, and buzzed angrily at Tanhony! He was led off te phremises by Franklin Briner te Chief Engineer. THE END"

**Comment by pali6** [comment]
I feel like it's maybe a bit too subtle?

File: code/obj/item/clothing/shoes.dm [hunk 4]
```diff
@@ -382,6 +382,25 @@
...
+	kick_bonus = 2 //kicking someone with wooden shoes has gotta hurt
+	step_sound = "step_wood"
+	step_priority = STEP_PRIORITY_LOW
+	setupProperties()
+		..()
+		setProperty("meleeprot", 1)
+
+	get_desc(var/dist, var/mob/user)
+		if (user.traitHolder.hasTrait("dutch"))
+			. += "Remembers you of the homeland."
```
Comment (pali6) [inline code/obj/item/clothing/shoes.dm:399]: I'm not a native speaker of English but to me this looks like it should be "Reminds" and not "Remembers". Ignore if I'm wrong or of the effect is intended.

Comment (ZeWaka) [reply code/obj/item/clothing/shoes.dm]: "Reminds" is the proper English.

**Comment by flrsh** [comment]
I'm assuming you're a native Dutch speaker right?

**Comment by Bierkraan** [comment]
> I'm assuming you're a native Dutch speaker right?

Yes, I am a native speaker

File: code/obj/item/clothing/shoes.dm [hunk 5]
```diff
@@ -382,6 +382,25 @@
...
+	setupProperties()
+		..()
+		setProperty("meleeprot", 1)
+
+	get_desc(var/dist, var/mob/user)
+		if (user.traitHolder.hasTrait("dutch"))
+			. += "Reminds you of the beatiful space dikes back at home."
+		else
+			. += "A pair of shoes from someone far under sea level."
+		. += ..()
```
Comment (ZeWaka) [inline code/obj/item/clothing/shoes.dm:402]: like I said before, you probably want . = ..() first

**Comment by BatElite** [comment]
I think I get what you (& taaltutor) are trying with the p -> ph substitution , but it's going to look like a lot of P sounds are being replaced with F and that's not a thing.
(Also appeal -> aphpheal in Pali's examples just looks silly.)

**Comment by Bierkraan** [comment]
Will nu PR be merged?

**Comment by flrsh** [comment]
If other people want to merge it yes.

**Review by ZeWaka — CHANGES_REQUESTED** [review]
please see my previous review.

**Comment by Bierkraan** [comment]
> please see my previous review.

I had already done that
`get_desc(var/dist, var/mob/user)
		. = ..()
		if (user.traitHolder.hasTrait("dutch"))
			. += "Reminds you of the beatiful space dikes back at home."
		else
			. += "A pair of shoes from someone far under sea level."`

**Comment by ZeWaka** [comment]
> > please see my previous review.
> 
> I had already done that
> `get_desc(var/dist, var/mob/user) . = ..() if (user.traitHolder.hasTrait("dutch")) . += "Reminds you of the beatiful space dikes back at home." else . += "A pair of shoes from someone far under sea level."`

That's not the code in this PR.

**Comment by Bierkraan** [comment]
is it now?

**Pull request closed without merging**